modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js

```diff
@@ -18,17 +20,85 @@ var callSystemInfo = rpc.declare({
 	method: 'info'
 });
 
+function showUpgradeModal(type, current_version, new_version) {
+	ui.showModal(_('New Firmware Available'), [
+		E('p', _('A new %s version of OpenWrt is available: %s').format(type, new_version)),
+		E('p', _('Your current version is: %s').format(current_version)),
+		E('p', _('Please check the https://firmware-selector.openwrt.org/ for the firmware upgrade image')),
+		E('div', { class: 'btn', click: ui.hideModal }, _('Close')),
+	]);
+};
+
+function checkDeviceAvailable(boardinfo, new_version) {
+	const profile_url = 'https://downloads.openwrt.org/releases/%s/targets/%s/profiles.json'.format(new_version, boardinfo.release.target);
+	return fetch(profile_url)
+		.then(response => response.json())
+		.then(data => {
+			// special case for x86 and armsr
+			if (Object.keys(data.profiles).length == 1 && Object.keys(data.profiles)[0] == "generic") {
+				return true;
+			}
+
+			for (var i = 0; i < data.profiles.length; i++) {
+				if (data.profiles[i].profile == boardinfo.board_name) {
+					return true;
+				}
+				for (var j = 0; j < data.profiles[i].supported_devices.length; j++) {
+					if (data.profiles[i].supported_devices[j] == boardinfo.board_name) {
+						return true;
+					}
+				}
+			}
+		})
+		.catch(error => {
+			console.error('Failed to fetch profile information:', error);
```

Comment:
consider adding more context to the error. A user who doesn't know about this feature might be confused by this err string, even though it's a console message.